api.planx.uk/admin/session/oneAppXML.ts

```diff
@@ -0,0 +1,42 @@
+import { gql } from "graphql-request";
+import { Request, Response, NextFunction } from "express";
+import { adminGraphQLClient } from "../../hasura";
+
+export const getOneAppXML = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const xml = await fetchSessionXML(req.params.sessionId);
+    res.set("content-type", "text/xml")
+    return res.send(xml)
+  } catch (error) {
+    return next({ message: "Failed to get OneApp XML: " + (error as Error).message });
+  }
+};
+
+const fetchSessionXML = async (sessionId: string) => {
+  try {
+    const client = adminGraphQLClient;
```

Comment:
nit: I've been slowing migrating to and appreciating this pattern which feels a bit cleaner to me: `import { adminGraphQLClient as client } from "../../hasura"`

Reply:
That's much neater - thanks.